### PR TITLE
Catch long username to error silently

### PR DESCRIPTION
**Pull request by georgeschena**

### CODE REVIEW

File: lib/logging/post_auth.rb [hunk 1]
```diff
@@ -13,7 +13,15 @@ def execute(params:)
...
-      Session.create(session_params.merge(username: username.to_s.upcase))
+      if username.to_s.length > 6
+        malformed_username
+      else
+        Session.create(session_params.merge(username: username.to_s.upcase))
+      end
+    end
+
+    def malformed_username
+      puts 'username that was used is larger than 6 characters and has errored silently'
```
Comment (emileswarts) [inline lib/logging/post_auth.rb]: "#{username} was invalid, logging rejected"

File: lib/logging/post_auth.rb [hunk 2]
```diff
@@ -13,7 +13,15 @@ def execute(params:)
     VALID_MAC_LENGTH = 17
 
     def create_user_session
-      Session.create(session_params.merge(username: username.to_s.upcase))
+      if username.to_s.length > 6
```
Comment (emileswarts) [inline lib/logging/post_auth.rb]: invalid_username?(username)

File: lib/logging/post_auth.rb [hunk 3]
```diff
@@ -13,7 +13,15 @@ def execute(params:)
...
+        malformed_username
```
Comment (emileswarts) [inline lib/logging/post_auth.rb]: handle_invalid_username

File: spec/features/post_auth_spec.rb [hunk 4]
```diff
@@ -187,4 +187,11 @@
       end
     end
   end
+
+  context 'given a username longer than 6 characters' do
+    it 'stops the error from blowing up' do
+      username = "very_long_username"
+      get "/logging/post-auth/user/#{username}/cert-name//mac//ap//site//result/success"
```
Comment (emileswarts) [inline spec/features/post_auth_spec.rb]: expect { get "/logging/post-auth/user/#{username}/cert-name//mac//ap//site//result/success"}.to_not raise_error

File: lib/logging/post_auth.rb [hunk 5]
```diff
@@ -13,7 +13,19 @@ def execute(params:)
...
+        Session.create(session_params.merge(username: username.to_s.upcase))
+      end
+    end
+
+    def invalid_username?(username)
+      username.to_s.length > 6
+    end
+
+    def handle_invalid_username
+      puts "#{username} was invalid due to being longer than 6 characters, logging rejected"
```
Comment (cybojenix) [inline lib/logging/post_auth.rb]: It would be better to use `logger` here.
You can pass it through via the controller

**Pull request merged**